ocaml/xapi/sm.ml

```diff
@@ -276,23 +302,31 @@ let vdi_compose dconf driver sr vdi1 vdi2 =
   in
   Sm_exec.parse_unit (Sm_exec.exec_xmlrpc (driver_filename driver) call)
 
-let vdi_epoch_begin dconf driver sr vdi =
+let vdi_epoch_begin ?traceparent dconf driver sr vdi =
+  Tracing.with_tracing ?parent:(Option.some traceparent)
+    ~name:"SM.vdi_epoch_begin"
+  @@ fun _ ->
   debug "vdi_epoch_begin" driver
     (sprintf "sr=%s vdi=%s" (Ref.string_of sr) (Ref.string_of vdi)) ;
   let call =
     Sm_exec.make_call ~sr_ref:sr ~vdi_ref:vdi dconf "vdi_epoch_begin" []
   in
   Sm_exec.parse_unit (Sm_exec.exec_xmlrpc (driver_filename driver) call)
 
-let vdi_epoch_end dconf driver sr vdi =
+let vdi_epoch_end ?traceparent dconf driver sr vdi =
+  Tracing.with_tracing ?parent:(Option.some traceparent) ~name:"SM.epoch_end"
```

Comment:
```suggestion
  Tracing.with_tracing ?parent:(Option.some traceparent) ~name:"SM.vdi_epoch_end"
```

Reply:
Thanks, I missed that!

Reply:
https://github.com/xapi-project/xen-api/pull/5293/commits/81310de0ec3c3af9a8e00ed239b782c65556ea1b This should fix it.